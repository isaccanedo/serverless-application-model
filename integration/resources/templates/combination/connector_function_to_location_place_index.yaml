Transform: AWS::Serverless-2016-10-31-test

Resources:
  TriggerFunction:
    Type: AWS::Serverless::Function
    Properties:
      Runtime: python3.8
      Handler: index.handler
      InlineCode: |
        import boto3
        import os
        client = boto3.client('location')

        def handler(event, context):
          indexName = os.environ['LOCATION_INDEX']
          response = client.describe_place_index(
            IndexName=indexName
          )
          response = client.search_place_index_for_position(
            IndexName=indexName,
            MaxResults=10,
            Language='en',
            Position=[
                -123.12,
                49.28
            ]
          )
          response = client.search_place_index_for_text(
            IndexName=indexName,
            Text="effiel tow"
          )
          response = client.search_place_index_for_suggestions(
            IndexName=indexName,
            Text="effiel tow"
          )
      Environment:
        Variables:
          LOCATION_INDEX: !Sub ${AWS::StackName}-PI


  MyPlace:
    Type: AWS::Location::PlaceIndex
    Properties:
      DataSource: Here
      IndexName: !Sub ${AWS::StackName}-PI

  MyConnector:
    Type: AWS::Serverless::Connector
    Properties:
      Source:
        Id: TriggerFunction
      Destination:
        Id: MyPlace
      Permissions:
      - Read
Metadata:
  SamTransformTest: true
