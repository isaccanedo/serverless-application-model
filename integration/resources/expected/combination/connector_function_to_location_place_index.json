[
  {
    "LogicalResourceId": "TriggerFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "TriggerFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "MyPlace",
    "ResourceType": "AWS::Location::PlaceIndex"
  },
  {
    "LogicalResourceId": "MyConnectorPolicy",
    "ResourceType": "AWS::IAM::ManagedPolicy"
  }
]
